compiler/src/iree/compiler/InputConversion/Common/AutoInputConversionPipeline.cpp

```diff
@@ -166,6 +167,36 @@ void AutoInputConversionPipelinePass::runOnOperation() {
   }
 }
 
+void AutoInputConversionPipelinePass::getDependentDialects(
+    DialectRegistry& registry) const {
+  // Register dialects from all possible pipelines, as we do not statically know
+  // which pipeline will be selected, while dialect registration happens before
+  // we run any detection on the input.
```

Comment:
Could you add a TODO that we should revise this? E.g., we are building pipelines just to discard them here and then rebuilding below. This may require different support upstream or some such.